src/Simplify.cpp

```diff
@@ -525,7 +538,11 @@ class Simplify : public IRMutator {
         const Select *select_a = a.as<Select>();
         const Select *select_b = b.as<Select>();
 
-        if (const_int(a, &ia) &&
+        if (is_error_call(call_a)) {
```

Comment:
> Signed overflow should cancel too: if x is signed_integer_overflow(y), 
> then x/x = 1, and x - x = 0.

To clarify: if x is the same instance of signed_integer_overflow(), right? i.e., two different instances most certainly shouldn't cancel. Is that the case now?